### ISSUE TITLE
role list with --job-template fails SSL check

### ISSUE THREAD

**Issue by phips** [ansible/tower-cli#243]
Specifying a `role list` action with --job-template fails SSL check (even when verify_ssl is set to False in tower-cli.cfg)…
```
[root@host site-packages]# tower-cli role list  -h https://tower/ -u user -p XXXXX --user 93 --job-template Practices-JenkinsMaster-Automation
Suggestion: add the correct http:// or https:// prefix to the host configuration.
Error: Could not get job_template. Could not establish a secure connection. Please add the server to your certificate authority.
```

…yet a list without --job-template switch works…
```
[root@host site-packages]# tower-cli role list  -h https://tower/ -u user -p XXXXX --user 93
==== ==================== ================================== =============
id          type                   resource_name            resource_type
==== ==================== ================================== =============
1 Admin                Practices-JenkinsMaster-Automation job template
```

**Comment by wenottingham** [comment]
Something is very weird with the parsing here of commandline arguments:

```
[notting@nostromo: ~]$ tower-cli role list -h https://tower-test.local/ -u admin -p a-password --user 1
=== ==================== ============================ ============= 
id          type                resource_name         resource_type 
=== ==================== ============================ ============= 
133 Admin                Apply standard configuration job template
114 Admin                Destroy cloud instances      job template
111 Admin                Provision cloud instances    job template
  1 System Administrator
  2 Admin                admin                        user
108 Admin                Deploy application           job template
 12 Admin                Demo Credential              credential
=== ==================== ============================ ============= 
[notting@nostromo: ~]$ tower-cli role list -h https://tower-test.local/ -u admin -p a-password --user 1 --job-template "Apply standard configuration" -v
Error: Could not get job_template. There was a network error of some kind trying to connect to Tower.

The most common  reason for this is a settings issue; is your "host" value in `tower-cli config` correct?
Right now it is: "127.0.0.1".
```

**Comment by AlanCoding** [comment]
I came up with a theory, tested it, and now I can say that its at the root of both this and #237.

Our config is unreliable, because it applies the static settings where one would expect to get runtime settings.

### Exhibit A

```
tower-cli config verbose false
tower-cli job_template list --project "Examples" -v
```

### Exhibit B

```
tower-cli config verbose true
tower-cli job_template list --project "Examples"
```

You would expect these to both print the same thing, right? Wrong. Exhibit A will not print the lookups for the related project field. Yet it will print debug messages in another place. This is because some dynamic imports are happening, causing the runtime settings to get squashed when deep in the bowels of the `Related` type code.

The same thing is happening in the examples presented above, and I feel fairly confident in saying that _the related lookup is happening on a different Tower server_ in those given examples.

**Comment by AlanCoding** [comment]
believed to be solved by #266

**Comment by jangsutsr** [comment]
Issue solved. Patch will ship with 3.1.1 release.
